userspace/libsinsp/test/sinsp.ut.cpp

```diff
@@ -456,3 +452,74 @@ TEST_F(sinsp_with_test_input, execveat_invalid_path)
 		FAIL();
 	}
 }
+
+// the following tests (test_file_name_...) check some corner cases in open/create files.
+// some versions of the bpf probe or the kmod may use a literal "<NA>" or "(NULL)" in case there is no information
+// available from the userspace buffers. Both these cases need to be properly handled, if no data is available upon
+// entry it needs to be taken from exit, otherwise the entry information should be trusted.
+
+TEST_F(sinsp_with_test_input, test_file_name_null)
```

Comment:
This test case seems really similar to `TEST_F(sinsp_with_test_input, enter_event_retrieval)` 2 are better than one but not sure it makes sense to keep both :thinking: WDYT? BTW, at a certain point in time, we should create different suites for different cases, maybe for different events like `sinsp_open_events` suite or `sinsp_link_events` otherwise we would obtain a huge file with tons of tests... Not sure this is the right PR to do this, but maybe it could be a next step :thinking:

Reply:
1. good point, this one tests a bit more things (three syscalls, more potentially troublesome cases and explicit `fd.name`) but perhaps it could be better to refactor this to test everything in the same place
2. for sure, I didn't want to commit to a specific split before having a few tests ready, but I expect to create multiple files in a single, easy to review, PR